core/src/wallet/tezos/TezosLikeAccount2.cpp

```diff
@@ -127,27 +127,32 @@ namespace ledger {
                         }
 
                         using TxsBulk = TezosLikeBlockchainExplorer::TransactionsBulk;
-                        static std::function<Future<Unit> (size_t)> getTxs = [self] (size_t id) -> Future<Unit> {
-                            std::vector<std::string> addresses{self->_originatedAccounts[id]->getAddress()};
-                            // TODO: Get info of last fetched block
-                            // For the moment we start synchro from the beginning
-                            return self->_explorer->getTransactions(addresses).flatMap<Unit>(self->getContext(), [=] (const std::shared_ptr<TxsBulk> &bulk) {
-                                auto uid = TezosLikeAccountDatabaseHelper::createOriginatedAccountUid(self->getAccountUid(), addresses[0]);
-                                vector::map<int, TezosLikeBlockchainExplorerTransaction>(bulk->transactions, [=] (const TezosLikeBlockchainExplorerTransaction &tx) {
-                                    soci::session sql(self->getWallet()->getDatabase()->getPool());
-                                    return self->putTransaction(sql, tx, uid, addresses[0]);
-                                });
 
-                                if (id == self->_originatedAccounts.size() - 1) {
-                                    return Future<Unit>::successful(Unit());
-                                }
-
-                                return getTxs(id + 1);
-                            }).recover(self->getContext(), [self] (const Exception& ex) -> Unit {
-                                throw ex;
-                            });
-                        };
-                        return getTxs(0);
+                        static std::function<Future<Unit> (std::shared_ptr<TezosLikeAccount>, size_t)> getTxs =
+                                [] (const std::shared_ptr<TezosLikeAccount> &account, size_t id) -> Future<Unit> {
```

Comment:
The `id` - given as parameter - is really useful ? Because all I see is a "opaque" check in line 146 and you can retrieve it from `_originatedAccounts` before create a new account. May I wrong but the logic around the `id` parameter isn't clear.